File: 2235-capitalize-the-title/capitalize-the-title.java
class Solution {
    public String capitalizeTitle(String title) {
        char[] arr = title.toCharArray();
        int n = arr.length;
        for(int i = 0;i<n;i++){
            int index = i;
        while(i < n && arr[i] != ' '){
            arr[i] = Character.toLowerCase(arr[i]);
            i++;
        }
        if(i - index > 2){
            arr[index] = Character.toUpperCase(arr[index]);
        }
        }
        return String.valueOf(arr);
}
}

File: 2235-capitalize-the-title/capitalize-the-title.py
class Solution:
    def capitalizeTitle(self, title: str) -> str:
        arr = list(title)
        n = len(arr)
        i = 0
        while i< n:
            index = i
            while i<n and arr[i] != ' ':
                arr[i] = arr[i].lower()
                i += 1
            if i - index >2:
                arr[index] = arr[index].upper()
            i+=1
        return ''.join(arr)